src/scoped.rs

```diff
@@ -220,32 +232,108 @@ impl<'a> fmt::Debug for Scope<'a> {
     }
 }
 
-impl<T> fmt::Debug for ScopedJoinHandle<T> {
+impl<'a, T> fmt::Debug for ScopedJoinHandle<'a, T> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         write!(f, "ScopedJoinHandle {{ ... }}")
     }
 }
 
 impl<'a> Scope<'a> {
+    /// Joins a spawned thread that is not joined yet.
```

Comment:
Maybe there should be more documentation on which order (reverse of `spawn`s or the same?) threads are joined in.